hikari/events/base_events.py

```diff
@@ -52,10 +52,28 @@
 NO_RECURSIVE_THROW_ATTR: typing.Final[str] = "___norecursivethrow___"
 
 
-@attr.define(kw_only=True, weakref_slot=False)
 class Event(abc.ABC):
     """Base event type that all Hikari events should subclass."""
 
+    __slots__: typing.Sequence[str] = ()
+
+    __dispatches: typing.ClassVar[typing.Tuple[typing.Type[Event], ...]]
+
+    def __init_subclass__(cls) -> None:
+        super().__init_subclass__()
+        # hasattr doesn't work with private variables in this case so we use a try except.
+        # We need to set Event's __dispatches when the first subclass is made as Event cannot
+        # be included in a tuple literal on itself due to not existing yet.
+        try:
+            Event.__dispatches
+        except AttributeError:
+            Event.__dispatches = (Event,)
+
+        mro = cls.mro()
+        # We don't have to explicitly include Event here as issubclass(Event, Event) returns True.
+        # Non-event classes should be ignored.
+        cls.__dispatches = tuple(cls for cls in mro if issubclass(cls, Event))
+
```

Comment:
I am starting to think this is not a good idea. Building the list when we need it (which is not a lot) seems to be much more efficient (memory) than storing it in all the event classes

Reply:
> Building the list when we need it (which is not a lot)
this list is used for every event dispatch though 🤔

Reply:
Completely missed that, nvm :+1: